optimum/utils/input_generators.py

```diff
@@ -1090,11 +1090,12 @@ def __init__(
             random_sequence_length_range=random_sequence_length_range,
             **kwargs,
         )
-        self.num_kv_heads = self.num_kv_heads = (
-            normalized_config.num_kv_heads
-            if (normalized_config.new_decoder_architecture or not normalized_config.multi_query)
-            else 1
-        )
+        if normalized_config.new_decoder_architecture and normalized_config.multi_query:
+            self.num_kv_heads = normalized_config.num_attention_heads
+        elif normalized_config.new_decoder_architecture and not normalized_config.multi_query:
+            self.num_kv_heads = normalized_config.num_kv_heads
+        else:
+            self.num_kv_heads = 1
```

Comment:
This can be simplified as follows I think:
```suggestion
        if normalized_config.new_decoder_architecture and normalized_config.multi_query:
            self.num_kv_heads = normalized_config.num_attention_heads
        elif normalized_config.new_decoder_architecture:
            self.num_kv_heads = normalized_config.num_kv_heads
        else:
            self.num_kv_heads = 1
```